frontend/app.js

```diff
@@ -0,0 +1,510 @@
+var React = require('react');
+var ReactDOM = require('react-dom');
+
+class Root extends React.Component {
+  constructor() {
+    super()
+
+    this.state = {
+      tasks: null,
+      taskName: '',
+      taskDescription: '',
+      childVisible: false,
+      submitBar: false,
+      completedState: 0
+    };
+  }
+
+  componentDidMount() {
+    fetch('/task')
+      .then((data) => {
+        return data.json()
+      })
+      .then((json) => {
+        this.setState({
+          tasks: json.tasks
+        })
+      })
+  }
+
+  addTask(event) {
+    event.preventDefault();
+    var name = this.state.taskName
+    var description = this.state.taskDescription
+    if (name == '') {
```

Comment:
If you introduce a validation on the client side, you should probably introduce it on the server side as well. At the end of the day, it's much more important that you validate your data on the server side anyway.